api/src/main/java/uk/gov/hmcts/payment/api/controllers/CasePaymentOrdersController.java

```diff
@@ -0,0 +1,94 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.payment.api.contract.CasePaymentOrdersDto;
+import uk.gov.hmcts.payment.api.dto.mapper.CasePaymentOrdersMapper;
+import uk.gov.hmcts.payment.api.service.CasePaymentOrdersService;
+import uk.gov.hmcts.payment.casepaymentorders.client.dto.CpoGetResponse;
+import uk.gov.hmcts.payment.casepaymentorders.client.exceptions.CpoBadRequestException;
+import uk.gov.hmcts.payment.casepaymentorders.client.exceptions.CpoClientException;
+import uk.gov.hmcts.payment.casepaymentorders.client.exceptions.CpoInternalServerErrorException;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+
+/**
+ * Case payment orders controller
+ */
+
+@RestController
+@Api(tags = {"Case Payment Order"})
+@SwaggerDefinition(tags = {@Tag(name = "CasePaymentOrderController", description = "Case Payment Order REST API")})
+public class CasePaymentOrdersController {
+    private static final Logger LOG = LoggerFactory.getLogger(CasePaymentOrdersController.class);
+
+    private final CasePaymentOrdersService casePaymentOrdersService;
+    private final CasePaymentOrdersMapper casePaymentOrdersMapper;
+
+    @Autowired
+    public CasePaymentOrdersController(CasePaymentOrdersService casePaymentOrdersService,
+                                       CasePaymentOrdersMapper casePaymentOrdersMapper) {
+        this.casePaymentOrdersService = casePaymentOrdersService;
+        this.casePaymentOrdersMapper = casePaymentOrdersMapper;
+    }
+
+    @ApiOperation(value = "Get payment orders for a case", notes = "Get payment orders for a case")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Payment orders retrieved"),
+        @ApiResponse(code = 400, message = "Bad request"),
+        @ApiResponse(code = 401, message = "Credentials are required to access this resource"),
+        @ApiResponse(code = 403, message = "Case Payment Orders info forbidden"),
+        @ApiResponse(code = 500, message = "Downstream system error")
+    })
+    @RequestMapping(value = "/case-payment-orders", method = GET)
+    public CasePaymentOrdersDto retrieveCasePaymentOrders(
+        @ApiParam(value = "Coma separated list of case ids.")
+        @RequestParam(name = "case-ids", required = false) String caseIds,
```

Comment:
Shouldn’t the case list be mandatory ?

Reply:
Could be. I have left the CpoServiceClient unchanged, meaning it accepts "case-ids" and "ids" as optional parameters. It is because the CPO has them optional and will return an empty list when none provided. Do you want to make it mandatory in the payment-app?

Reply:
I think we should have case-Ids as mandatory, also What happens if request with empty case list ?

Reply:
Changed case-ids to mandatory